### PR TITLE
perf(core): Minimize binary data deletion calls

### PR DESCRIPTION
**Pull request by ivov**
Quick idea, @krynble wdyt?

### CODE REVIEW

File: packages/cli/src/databases/migrations/common/1697198787123-AddExecutionHasBinaryData.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import type { MigrationContext, ReversibleMigration } from '@db/types';
+
+const tableName = 'execution_entity';
+
+export class AddExecutionHasBinaryData1697198787123 implements ReversibleMigration {
+	async up({ schemaBuilder: { addColumns, column } }: MigrationContext) {
+		await addColumns(tableName, [column('hasBinaryData').bool.default(false)]);
```
Comment (krynble) [inline packages/cli/src/databases/migrations/common/1697198787123-AddExecutionHasBinaryData.ts:7]: The problem I see is that we're not populating existing executions, effectively leaving those files iin the disk forever if using `filesystem` already

File: packages/cli/src/executionLifecycleHooks/hasBinaryData.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import type { IRun } from 'n8n-workflow';
+
+export function hasBinaryData(run: IRun) {
+	const { runData } = run.data.resultData;
+
+	return Object.keys(runData).some((nodeName) => {
```
Comment (krynble) [inline packages/cli/src/executionLifecycleHooks/hasBinaryData.ts:6]: My concern here is that we're looking at the first `runIndex` and only the `firstItem` of a node, and maybe we should iterate over all nodes and all run indices and check if _any_ contains binary data right?

Maybe a more assertive approach is to use the `NodeExecuteFunctions` helper functions and flag it whenever we "create" a binary file and flag it as part of the `packages/workflow/src/Workflow.ts` instance.

File: packages/cli/src/databases/migrations/common/1697198787123-AddExecutionHasBinaryData.ts (hunk 1 above)
Comment (ivov) [reply packages/cli/src/databases/migrations/common/1697198787123-AddExecutionHasBinaryData.ts:7]: Definitely! We'd need migrations, or make the skipping exclusive to `s3`.

File: packages/cli/src/executionLifecycleHooks/hasBinaryData.ts (hunk 2 above)
Comment (ivov) [reply packages/cli/src/executionLifecycleHooks/hasBinaryData.ts:6]: Good point, but to clarify this needs to be part of the execution, not the workflow.

**Comment by ivov** [comment]
Closing here, only a draft to explore an idea.

**Pull request closed without merging**